### PR TITLE
target/riscv: decrease modify_privilege function nesting level

### PR DESCRIPTION
**Pull request by fkhaidari**

### CODE REVIEW

**Comment by fkhaidari** [comment]
@JanMatCodasip, @MarekVCodasip, could you take a look please?

File: src/target/riscv/riscv-013.c [hunk 1]
```diff
@@ -3082,36 +3082,46 @@ static int read_sbcs_nonbusy(struct target *target, uint32_t *sbcs)
...
+	if (get_field(dcsr, CSR_DCSR_PRV) == PRV_M)
+		return ERROR_OK;
 
-			/* Write MSTATUS */
-			if (*mstatus != *mstatus_old)
-				if (register_write_direct(target, GDB_REGNO_MSTATUS, *mstatus) != ERROR_OK)
-					return ERROR_FAIL;
-		}
-	}
+	/* MPP = PRIV */
```
Comment (JanMatCodasip) [inline src/target/riscv/riscv-013.c]: Cosmetic detail:

```suggestion
	/* mstatus.mpp <- dcsr.prv */
```

File: src/target/riscv/riscv-013.c [hunk 2]
```diff
@@ -3082,36 +3082,46 @@ static int read_sbcs_nonbusy(struct target *target, uint32_t *sbcs)
...
+	*mstatus = set_field(*mstatus, MSTATUS_MPP, get_field(dcsr, CSR_DCSR_PRV));
 
-	return ERROR_OK;
+	/* MPRV = 1 */
```
Comment (JanMatCodasip) [inline src/target/riscv/riscv-013.c]: Cosmetic:

```suggestion
	/* mstatus.mprv <- 1 */
```

File: src/target/riscv/riscv-013.c [hunk 3]
```diff
@@ -3082,36 +3082,46 @@ static int read_sbcs_nonbusy(struct target *target, uint32_t *sbcs)
...
+/* TODO: return mem_access_result_t */
 static int modify_privilege(struct target *target, uint64_t *mstatus, uint64_t *mstatus_old)
 {
-	if (riscv_virt2phys_mode_is_hw(target)
-			&& has_sufficient_progbuf(target, 5)) {
-		/* Read DCSR */
-		uint64_t dcsr;
-		if (register_read_direct(target, &dcsr, GDB_REGNO_DCSR) != ERROR_OK)
-			return ERROR_FAIL;
+	if (!riscv_virt2phys_mode_is_hw(target))
```
Comment (JanMatCodasip) [inline src/target/riscv/riscv-013.c:3100]: Detail, if you like:

```suggestion
	assert(mstatus);
	assert(mstatus_old);

	if (!riscv_virt2phys_mode_is_hw(target))
```

**Review by JanMatCodasip — DISMISSED** [review]
LGTM, thanks.

I am sending just few optional cosmetic suggestions.

File: src/target/riscv/riscv-013.c (hunk 1 above)
Comment (fkhaidari) [reply src/target/riscv/riscv-013.c]: Addressed

File: src/target/riscv/riscv-013.c (hunk 2 above)
Comment (fkhaidari) [reply src/target/riscv/riscv-013.c]: Addressed

File: src/target/riscv/riscv-013.c (hunk 3 above)
Comment (fkhaidari) [reply src/target/riscv/riscv-013.c:3100]: Addressed

**Review by JanMatCodasip — APPROVED** [review]
LGTM

**Review by en-sc — APPROVED** [review]

**Pull request merged**